### ISSUE TITLE
Switching with ctr-shift-alt forward/backward doesn't work correctly with XMETHOD xorg 

### ISSUE THREAD

**Issue by divx118** [dnschneid/crouton#1475]
```
chronos@localhost / $ sudo enter-chroot -n trusty croutonversion
Entering /mnt/stateful_partition/crouton/chroots/trusty...
Unknown username "whoopsie" in message bus configuration file
WARNING: /etc/rc.local failed with code 1
crouton: version 1-20150304145447~master:550afdb3
release: trusty
architecture: amd64
xmethod: xiwi
targets: xiwi,xorg,xfce,keyboard,extension,chrome
host: version 6812.3.0 (Official Build) dev-channel falco 
Unmounting /mnt/stateful_partition/crouton/chroots/trusty...
```

There is an unknown display added which isn't active. So when using the hotkeys mouse and keyboard switches away, but display will still show chromeos or chroot.  
If you logout in the chroot you will end up with a chrome window that is not responsive. Current workaround is to first switch to chromeos and use ctr-c in the crosh to end the chroot.

Some outputs with no chroot windows started:

```
chronos@localhost / $ sudo enter-chroot -n trusty host-x11 croutonwmtools list nim
Password: 
Entering /mnt/stateful_partition/crouton/chroots/trusty...
Unknown username "whoopsie" in message bus configuration file
aura_root_0 2097156*
Unknown 6291457
Unmounting /mnt/stateful_partition/crouton/chroots/trusty...
```

`sudo enter-chroot -n trusty croutonwebsocket` will only show "cross Chrome OS" in the extension.

**Comment by divx118** [comment]
Update: This is also an issue when using xiwi. Although the extension doesn't list "Unknown 6291457" it still tries to switch to it.  
Probably this is only affecting devices that didn't make the switch to freon yet.

**Comment by divx118** [comment]
It seems it is correct to say that this is only affecting devices that didn't make the switch to freon yet.
My HP chromebook 14 just updated on dev-channel and made the switch to freon.  
Now I don't have the problem anymore.
```
chronos@localhost / $ sudo enter-chroot -n trusty croutonversion
Entering /mnt/stateful_partition/crouton/chroots/trusty...
Unknown username "whoopsie" in message bus configuration file
crouton: version 1-20150304145447~master:550afdb3
release: trusty
architecture: amd64
xmethod: xiwi
targets: xiwi,xorg,xfce,keyboard,extension,chrome
host: version 6812.29.0 (Official Build) dev-channel falco 
Unmounting /mnt/stateful_partition/crouton/chroots/trusty...
```

**Comment by drinkcat** [comment]
Not sure what's going on: it works fine on my non-freon peppy:

```
crouton: version git
release: trusty
architecture: amd64
xmethod: xorg
targets: xorg,xiwi,keyboard,xfce
host: version 6680.64.0 (Official Build) stable-channel peppy 
kernel: Linux localhost 3.8.11 #1 SMP Sat Mar 7 06:52:13 PST 2015 x86_64 x86_64 x86_64 GNU/Linux
freon: no
```

**Comment by divx118** [comment]
Well I can't reproduce it anymore, since my device switched to freon I don't have that problem. Probably not something to worry about, until there will be more reports.

**Comment by drinkcat** [comment]
Okay, let's close it for now. Thanks.

**Comment by penntaylor** [comment]
Coming here from #1678, definitely same issue.
```
crouton: version 1-20150423153254~master:2f55723c
release: trusty
architecture: amd64
xmethod: xorg
targets: keyboard,extension,gnome
host: version 6812.75.2 (Official Build) stable-channel swanky 
kernel: Linux localhost 3.10.18 #1 SMP Tue Apr 14 20:43:12 PDT 2015 x86_64 x86_64 x86_64 GNU/Linux
freon: no
```
I've noticed that in addition to whatever is causing this extra unknown display problem, croutontriggerd and xbindkeys seem to step on each other. I killed off xbindkeys and altered croutontriggerd to use ctrl+esc as the display cycle key combination. At the very least it makes having to press the key sequence 2 or 3 times to get from chromeos to the chroot a less obnoxious experience.

**Comment by rmkraus** [comment]
I also get this issue sporadically. Most of the time, one of these issues will creep up. Every once in a while, everything works just fine. I know how much work went into this project so I just wanted to say thank you. It is very cool, even if it is acting up a bit for me right now.

```bash
crouton: version 1-20150423153254~master:2f55723c
release: trusty
architecture: amd64
xmethod: xiwi
targets: xiwi,xorg,x11,kde
host: version 6812.75.2 (Official Build) stable-channel quawks 
kernel: Linux localhost 3.10.18 #1 SMP Tue Apr 14 20:45:15 PDT 2015 x86_64 x86_64 x86_64 GNU/Linux
freon: no
```

**Comment by penntaylor** [comment]
**Possible** solution:
Change lines 109 and 110* of croutoncycle  from:
```
winlist="`host-x11 croutonwmtools list nim | \
              sort | awk '{ printf $NF " " }'`"
```
to
```
winlist="`host-x11 croutonwmtools list nim | \
              awk '$1 == "aura_root_0" { printf $NF }'`"
```
and the switching problem goes away: a single pass through croutoncycle will get you from chromeos into the chroot. The proposed change simply causes croutoncycle to ignore all windows besides the one containing aura_root_0. I can confirm that this also works fine with multiple chroot graphical sessions running, since those get individual display numbers rather than windows.

I am not cloning and issuing a pull request because I do not understand enough about the windows in the winlist to know whether this is a proper fix. I just know it solves the issue for me. As far as I can tell, the conditional branch containing these lines will never be executed on freon systems.

\* as per repo line numbers; on a live installation these become lines 311 and 312 (or thereabouts)

**Comment by penntaylor** [comment]
The above possible solution also appears to fix the other issue reported in the initial report:

"If you logout in the chroot you will end up with a chrome window that is not responsive. Current workaround is to first switch to chromeos and use ctr-c in the crosh to end the chroot."

**Comment by dnschneid** [comment]
I guess it's never too early to remove arbitrary host-x11 support, now that we have xiwi...

**Comment by Sebb767** [comment]
For some reason, the above fix didn't work for me, it still reported one unkown display, but one was gone. A maybe-not-that-clean fix for me was to reverse grep all unkown displays from winlist in croutoncycle:

```
    winlist="`host-x11 croutonwmtools list nim | \
              grep -vi "unknown" | \
              awk '$1 == "aura_root_0" { printf $NF }'`"
```

Hope that helps someone :)

**Comment by JohnAllanSmith** [comment]
I tried both penntaylor and Sebb767 suggestion on changes to croutoncycle but neither helped. I CAN shift from X to Chromebook, but not the other way around. So I have to CTRL+C the terminal

host: version 6812.88.0 (Official Build) stable-channel swanky 
kernel: Linux localhost 3.10.18 #1 SMP Fri May 8 16:05:18 PDT 2015 x86_64 GNU/Linux
freon: no
(II) AIGLX: Suspending AIGLX clients for VT switch
^C/usr/bin/xinit: connection to X server lost

waiting for X server to shut down Hangup
Hangup
Running exit commands...
(EE) Server terminated successfully (0). Closing log file.


Not sure if there's anything else I can try?
